### PR TITLE
First set of files to be updated using the mcpan shortcode.

### PR DESCRIPTION
**Pull request by wpr-curly2000**
Hi Brian,

As you proposed, I have updated 2 files with the mcpan shortcode:

legacy/_pub_2003_04_17_filters.md and legacy/_pub_2003_07_08_mod_perl.md

Since you cannot attach '.md' files directly.  I have appended a '.txt' to each file.  Pls let me know if there is an easier way to do attach these files.  
[_pub_2003_07_08_mod_perl.md.txt](https://github.com/tpf/perldotcom/files/6047063/_pub_2003_07_08_mod_perl.md.txt)
[_pub_2003_04_17_filters.md.txt](https://github.com/tpf/perldotcom/files/6047064/_pub_2003_04_17_filters.md.txt)

Thanks, Jeff.

### CODE REVIEW

**Comment by briandfoy** [comment]
Thanks for doing more work. But, we need to adjust how you do this. There are 41 changed files in this PR and that's unwieldy to handle. Next time, choose one or two files at a time.

File: content/legacy/_pub_1999_10_p5pdigest_THISWEEK-19991024.md [hunk 1]
```diff
@@ -196,7 +196,7 @@ Actually the real birthday was on 17 October, 1994, but there is an error in `pe
 
 ### <span id="Unicode_Character_Classes_Revisi">Unicode Character Classes Revisited</span>
 
-[Last week there was discussion of use of Unicode properties to define regex character classes](/pub/1999/10/p5pdigest/THISWEEK-19991017.html#charclasses). People interested should also consider reading the [Unicode Regular Expression Guidelines](http://www.unicode.org/unicode/reports/tr18/).
+[Last week there was discussion of use of Unicode properties to define regex character classes](/pub/1999/10/p5pdigest/THISWEEK-19991017.html#charclasses). People interested should also consider reading the [Unicode Regular Expression Guidelines](http://unicode.org/reports/tr18/tr18-5.1.html).
```
Comment (briandfoy) [inline content/legacy/_pub_1999_10_p5pdigest_THISWEEK-19991024.md:199]: This should be https://unicode.org/reports/tr18/

File: content/legacy/_pub_2000_10_p5pdigest_THISWEEK-20001030.md [hunk 2]
```diff
@@ -81,7 +81,7 @@ Work on the `Encode` module to convert character sets continues, and it's really
 
 Peter Prymmer's did an excellent job and created an EBCDIC-&gt;Unicode mapping with it; Nick came up with a POD translation of the documentation on [how Encode's mapping files work](http://www.xray.mpe.mpg.de/mailing-lists/perl5-porters/2000-10/msg01122.html). (We stole them from Tcl! Ha!)
 
-The thread wandered off into discussion of what the Unicode characters `0x0000` and `0xFFFF` mean. Don't just guess, see the [Unicode FAQ](http://www.unicode.org/unicode/faq/)!
+The thread wandered off into discussion of what the Unicode characters `0x0000` and `0xFFFF` mean. Don't just guess, see the [Unicode FAQ](https://unicode.org//faq/)!
```
Comment (briandfoy) [inline content/legacy/_pub_2000_10_p5pdigest_THISWEEK-20001030.md:84]: This should be https://unicode.org/faq/

File: content/legacy/_pub_2000_12_net.md [hunk 3]
```diff
@@ -59,7 +59,7 @@ Here are some references for Web services:
 
 -   <http://www.xml-rpc.com>
 
--   <http://www.soaprpc.com>
+-   <http://web.archive.org/web/20190305142256/http://www.soaprpc.com:80/>
```
Comment (briandfoy) [inline content/legacy/_pub_2000_12_net.md:62]: Turn this whole section into markdown links.

File: content/legacy/_pub_2001_06_p5pdigest_THISWEEK-20010612.md [hunk 4]
```diff
@@ -47,7 +47,7 @@ Note that at the moment a few people already have Perforce access, such as Gurus
 
 ### <span id="Regex_Negation">Regex Negation</span>
 
-Jeff Pinyan noted that a new Java regex package contained support for the following regular expression negation: `[\w'\-[^\d]]`, which matches any word character, apostrophe, or hyphen, EXCEPT digits, and asked whether support was planned for Perl. He was pointed to the [Unicode Regular Expression Guidelines](http://www.unicode.org/unicode/reports/tr18/), which proposed a syntax: `[AEIOU[^A]]`, but was rather unclear on many points. The backwards compatibility police made an appearance, but otherwise nothing was resolved.
+Jeff Pinyan noted that a new Java regex package contained support for the following regular expression negation: `[\w'\-[^\d]]`, which matches any word character, apostrophe, or hyphen, EXCEPT digits, and asked whether support was planned for Perl. He was pointed to the [Unicode Regular Expression Guidelines](http://www.unicode.org/reports/tr18/tr18-20.html), which proposed a syntax: `[AEIOU[^A]]`, but was rather unclear on many points. The backwards compatibility police made an appearance, but otherwise nothing was resolved.
```
Comment (briandfoy) [inline content/legacy/_pub_2001_06_p5pdigest_THISWEEK-20010612.md:50]: This should be https://unicode.org/reports/tr18/

File: content/legacy/_pub_2001_08_p5pdigest_20010815.md [hunk 5]
```diff
@@ -72,7 +72,7 @@ Sarathy suggested we mandate POD to be in Unicode, and move towards assuming all
 
 ### <span id="Unicode_Normalisation">Unicode Normalisation</span>
 
-And while we're on the subject of Unicode, Sadahiro Tomoyuki rocks my world. Oh my. Not only has he produced an alternative and much cleaner (and more complete) module to handle normalization of Unicode data, he achieved the near-impossible and implemented the Unicode collation algorithm detailed in [Unicode Technical Report 10](http://www.unicode.org/unicode/reports/tr10/). This is a major bonus, since it allows us to correctly compare and store Unicode strings.
+And while we're on the subject of Unicode, Sadahiro Tomoyuki rocks my world. Oh my. Not only has he produced an alternative and much cleaner (and more complete) module to handle normalization of Unicode data, he achieved the near-impossible and implemented the Unicode collation algorithm detailed in [Unicode Technical Report 10](http://web.archive.org/web/20050323094117/http://www.unicode.org:80/unicode/reports/tr10/). This is a major bonus, since it allows us to correctly compare and store Unicode strings.
```
Comment (briandfoy) [inline content/legacy/_pub_2001_08_p5pdigest_20010815.md:75]: This should be https://unicode.org/reports/tr10/

File: content/legacy/_pub_2001_09_26_crypto1.md [hunk 6]
```diff
@@ -285,7 +285,7 @@ Meanwhile, if Eve has sniffed the network traffic to intercept the message and s
 <br />
 • <a href="http://www.ietf.org/rfc/rfc2437.txt">Kaliski &amp; Staddon, PCKS #1, RSA Cryptography Specifications</a><br />
 <br />
-• <a href="http://www.rsasecurity.com/rsalabs/pkcs/pkcs-3/">PKCS #3, Diffie-Hellman Key Agreement Standard</a><br />
+• <a href="http://web.archive.org/web/20061207062720/http://www.rsasecurity.com:80/rsalabs/pkcs/pkcs-3/">PKCS #3, Diffie-Hellman Key Agreement Standard</a><br />
```
Comment (briandfoy) [inline content/legacy/_pub_2001_09_26_crypto1.md:288]: This should be https://www.teletrust.de/fileadmin/files/oid/oid_pkcs-3v1-4.pdf

File: content/legacy/_pub_2001_10_p5pdigest_20011021.md [hunk 7]
```diff
@@ -67,7 +67,7 @@ Jarkko posted a [list of things to do](http://www.xray.mpe.mpg.de/mailing-lists/
...
 
-Jarkko, apparently working all alone, has made considerable progress towards making Perl comply with "Level 1 - Basic Unicode Support", as defined by the Unicode standard, [TR\#18 "Unicode Regular Expression Guidelines"](http://www.unicode.org/unicode/reports/tr18/). Some features remain unimplemented, though, including character class subtraction and end-of-line matching.
+Jarkko, apparently working all alone, has made considerable progress towards making Perl comply with "Level 1 - Basic Unicode Support", as defined by the Unicode standard, [TR\#18 "Unicode Regular Expression Guidelines"](http://web.archive.org/web/20050403115502/http://www.unicode.org:80/unicode/reports/tr18/). Some features remain unimplemented, though, including character class subtraction and end-of-line matching.
```
Comment (briandfoy) [inline content/legacy/_pub_2001_10_p5pdigest_20011021.md:70]: This should be https://unicode.org/reports/tr18/

File: content/legacy/_pub_2001_12_19_xmlrpc.md [hunk 8]
```diff
@@ -291,4 +291,4 @@ So, if you're a Perl programmer, and are looking to leveraging off a service tha
 
 <http://www.activestate.com>
 
-<http://xmlrpc-c.sourceforge.net/xmlrpc-howto/xmlrpc-howto.html>
+<http://web.archive.org/web/20051102170445/http://xmlrpc-c.sourceforge.net:80/xmlrpc-howto/xmlrpc-howto.html>
```
Comment (briandfoy) [inline content/legacy/_pub_2001_12_19_xmlrpc.md:294]: Turn this whole section in markdown links

File: content/legacy/_pub_2001_12_p6pdigest_20011208.md [hunk 9]
```diff
@@ -35,11 +35,11 @@ Parrot 0.03 should now be available. From the [release notes](http://archive.dev
 >
 > That's not all, however... We've more documentation, examples, a minature Scheme implementation, support for pushing and popping onto register and global user stacks, subroutine support for Jako, better platform support, and a whole lot more. See the NEWS and ChangeLog files for the rest.
 
-[Download](http://www.cpan.org/authors/id/S/SI/SIMON/parrot-0.0.3.tar.gz) it and give it a try.
+[Download](http://web.archive.org/web/20060619185554/http://www.cpan.org:80/authors/id/S/SI/SIMON/parrot-0.0.3.tar.gz) it and give it a try.
```
Comment (briandfoy) [inline content/legacy/_pub_2001_12_p6pdigest_20011208.md:38]: I'd change this to https://github.com/parrot/parrot

File: content/legacy/_pub_2002_08_22_exegesis5.md [hunk 10]
```diff
@@ -99,7 +99,7 @@ As Piers Cawley has so eloquently misquoted: *“It is a truth universally ackno
 
 ### <span id="#whats_the_diff">What's the diff?</span>
 
-So let's take a look at some of those new features. To do that, we'll consider a series of examples structured around a common theme: recognizing and manipulating data in the Unix *[diff](http://www.gnu.org/manual/diffutils-2.8.1/html_node/Detailed-Normal.html)*
+So let's take a look at some of those new features. To do that, we'll consider a series of examples structured around a common theme: recognizing and manipulating data in the Unix *[diff](http://web.archive.org/web/20031204104626/http://www.gnu.org:80/manual/diffutils-2.8.1/html_node/Detailed-Normal.html)*
```
Comment (briandfoy) [inline content/legacy/_pub_2002_08_22_exegesis5.md:102]: This should be https://www.gnu.org/software/diffutils/

File: content/legacy/_pub_2002_09_17_ewispp.md [hunk 11]
```diff
@@ -246,7 +246,7 @@ It is outside the scope of this article to cover graph operations, but I've name
 
     How is the Web browser going to communicate back to the Web server? The only way it can is by requesting pages (via URLs.) For a graph browser we need two different kinds.
 
-    First, an image representing the graph. Second, a HTML page containing the **IMG** tag for the graph and the [imagemap](http://www.cris.com/~automata/tutorial.shtml).
+    First, an image representing the graph. Second, a HTML page containing the **IMG** tag for the graph and the [imagemap](http://web.archive.org/web/20110112012108/http://www.cris.com:80/~automata/tutorial.shtml).
```
Comment (briandfoy) [inline content/legacy/_pub_2002_09_17_ewispp.md:249]: Let's make that one https://www.w3schools.com/html/html_images_imagemap.asp

File: content/legacy/_pub_2003_08_p6pdigest_20030803.md [hunk 12]
```diff
@@ -94,7 +94,7 @@ It turns out that Leo already has an "unportable, ugly, slightly tested" script
 
 [http://groups.google.com/groups](http://groups.google.com/groups?threadm=a05210610bb4c910e2831%40%5B63.120.19.221%5D)
 
-<http://www.hitchhiker.org/parrot_coverage> -- Josh's reports
+<http://web.archive.org/web/20050816082346/http://www.hitchhiker.org:80/parrot_coverage/> -- Josh's reports
```
Comment (briandfoy) [inline content/legacy/_pub_2003_08_p6pdigest_20030803.md:97]: Let's make all those markdown links

File: content/legacy/_pub_2003_08_p6pdigest_20030803.md [hunk 13]
```diff
@@ -120,7 +120,7 @@ Just as we were all giving Michal some good Wow, Joseph Ryan announced that he h
 
 [http://groups.google.com/groups](http://groups.google.com/groups?threadm=3F2A11BF.9000906%40osu.edu)
 
-<http://jryan.perlmonk.org/images/jirate.tar.gz>
+<http://web.archive.org/web/20170621155426/http://jryan.perlmonk.org/images/jirate.tar.gz>
```
Comment (briandfoy) [inline content/legacy/_pub_2003_08_p6pdigest_20030803.md:123]: convert to markdown links

File: content/legacy/_pub_2004_10_21_taint_testing_kata.md [hunk 14]
```diff
@@ -26,7 +26,7 @@ To be a better programmer, practice programming.
 
 It's not enough to practice, though. You must practice well and persistently. You need to explore branches and ideas and combinations as they come to your attention. Set aside some time to experiment with a new idea to see what you can learn and what you can use in your normal programming.
 
-How do you find new ideas? One way is through [code katas](https://pragprog.com/pragdave/Practices/CodeKata.rdoc), short pieces of code that start your learning.
+How do you find new ideas? One way is through [code katas](http://web.archive.org/web/20080705104855/http://pragprog.com/pragdave/Practices/CodeKata.rdoc), short pieces of code that start your learning.
```
Comment (briandfoy) [inline content/legacy/_pub_2004_10_21_taint_testing_kata.md:29]: http://codekata.com

File: content/legacy/_pub_2005_05_26_word_control.md [hunk 15]
```diff
@@ -20,7 +20,7 @@
...
+In a recent lightning article, [Customizing Emacs with Perl](/pub/2005/03/31/lightning2.html), [Bob DuCharme](/authors/bob-ducharme) explained how to use the Emacs `shell-command-on-region` function to invoke a Perl script on a marked region of text. Bob writes that he was reluctant to invest the time needed to write the [elisp](http://web.archive.org/web/20101228163508/http://www.gnu.org:80/software/emacs/elisp-manual/html_mono/elisp.html) code needed for a particular string manipulation, especially when he knew how much easier it would be for him to do that manipulation with Perl. However, by using the Emacs function `shell-command-on-region`, Bob could have his cake an eat it too--keep editing with Emacs, while using Perl on demand for string manipulation.
```
Comment (briandfoy) [inline content/legacy/_pub_2005_05_26_word_control.md:23]: https://www.gnu.org/software/emacs/manual/html_node/elisp/

File: content/legacy/_pub_2005_07_07_navwidgets.md [hunk 16]
```diff
@@ -35,7 +35,7 @@ Usually, the items in the navigation menus are a tree structure (as is the case
 
 Many sites provide links to traverse the pages of the site in order: a *Next* link to go to the next page, a *Previous* link to go to the previous page, an *Up* link to go to the section containing the current page, a *Contents* link to go to the main page, and so on.
 
-HTML can represent these links by using `<head>` tag and `<link rel="next" href="" />`. directives. Mozilla, Firefox, and Opera all support these tags. They can be also visible in the HTML as normal links, as is the case with [GNU Documentation](http://www.gnu.org/software/make/manual/html_node/make_toc.html).
+HTML can represent these links by using `<head>` tag and `<link rel="next" href="" />`. directives. Mozilla, Firefox, and Opera all support these tags. They can be also visible in the HTML as normal links, as is the case with [GNU Documentation](http://web.archive.org/web/20060204124456/http://www.gnu.org:80/software/make/manual/html_node/make_toc.html).
```
Comment (briandfoy) [inline content/legacy/_pub_2005_07_07_navwidgets.md:38]: https://ftp.gnu.org/old-gnu/Manuals/make-3.80/html_node/make_toc.html

File: content/legacy/_pub_2005_08_11_win32guitest.md [hunk 17]
```diff
@@ -68,7 +68,7 @@ First look for the main window of the application, and then descend the hierarch
 
 How can you know the windows hierarchy? There are two main ways. If you have written the GUI yourself or have access to its sources and have enough experience, you may find out what the hierarchy of windows is. Unfortunately, that's quite tricky and prone to error.
 
-Another much simpler way to do this on Windows platforms is to use the free [WinSpy++](http://www.catch22.net/software/winspy.asp) program. Basically, it allows you to [peek at an application's window structure](http://www.piotrkaluski.com/files/winguitest/docs/ch02.html).
+Another much simpler way to do this on Windows platforms is to use the free [WinSpy++](http://web.archive.org/web/20171023181503/http://www.catch22.net:80/software/winspy.asp) program. Basically, it allows you to [peek at an application's window structure](http://www.piotrkaluski.com/files/winguitest/docs/ch02.html).
```
Comment (briandfoy) [inline content/legacy/_pub_2005_08_11_win32guitest.md:71]: http://www.catch22.net/software/winspy

File: content/legacy/_pub_2005_08_18_ical_dot.md [hunk 18]
```diff
@@ -129,7 +129,7 @@ While this example does a good job of showing how to fill the data structure, it
 
 ### The Dot Specification
 
-[Dot](http://www.graphviz.org/Documentation/dotguide.pdf) (PDF) is a diagramming, or directed, graph language created by Emden Gansner, Eleftherios Koutsofios, and Stephen North at Bell Labs. There are several implementations of Dot, including [GraphViz](http://www.graphviz.org), [WebDot](http://www.graphviz.org/webdot/), and [Grappa](http://www.research.att.com/~john/Grappa/). Interestingly, [OmniGraffle](http://www.omnigroup.com/applications/omnigraffle/), a powerful diagramming tool for Macintosh computers, can read simple Dot files.
+[Dot](http://www.graphviz.org/Documentation/dotguide.pdf) (PDF) is a diagramming, or directed, graph language created by Emden Gansner, Eleftherios Koutsofios, and Stephen North at Bell Labs. There are several implementations of Dot, including [GraphViz](http://www.graphviz.org), [WebDot](http://web.archive.org/web/20171026075454/http://www.graphviz.org:80/webdot/), and [Grappa](http://www.research.att.com/~john/Grappa/). Interestingly, [OmniGraffle](http://www.omnigroup.com/applications/omnigraffle/), a powerful diagramming tool for Macintosh computers, can read simple Dot files.
```
Comment (briandfoy) [inline content/legacy/_pub_2005_08_18_ical_dot.md:132]: https://gitlab.com/graphviz/webdot

File: content/legacy/_pub_2005_10_20_scriptome.md [hunk 19]
```diff
@@ -47,7 +47,7 @@ When I told Neeraj's story to others in our group, they said that they have stru
 
 ### Harnessing the Power of the Atom
 
-[The Scriptome](http://sysbio.harvard.edu/csb/resources/computational/scriptome/) is a cookbook for munging biological data. The cookbook model nicely fits the UNIX paradigm of small tools that do simple operations. Instead of UNIX pipes, though, we encourage the use of intermediate files to avoid errors.
+[The Scriptome](http://web.archive.org/web/20141013192711/http://sysbio.harvard.edu/csb/resources/computational/scriptome/) is a cookbook for munging biological data. The cookbook model nicely fits the UNIX paradigm of small tools that do simple operations. Instead of UNIX pipes, though, we encourage the use of intermediate files to avoid errors.
```
Comment (briandfoy) [inline content/legacy/_pub_2005_10_20_scriptome.md:50]: http://archive.sysbio.harvard.edu/CSB/resources/computational/scriptome/UNIX/

File: content/promoting-perl-community-articles.md [hunk 20]
```diff
@@ -0,0 +1,55 @@
+{
```
Comment (briandfoy) [inline content/promoting-perl-community-articles.md:1]: What happened with this file that it's completely re-added?

File: _pub_2003_04_17_filters.md [hunk 21]
```diff
@@ -0,0 +1,284 @@
+{
```
Comment (briandfoy) [inline _pub_2003_04_17_filters.md:1]: What happened with this file that it's completely re-added?

**Review by briandfoy — CHANGES_REQUESTED** [review]

**Review by briandfoy — CHANGES_REQUESTED** [review]
There are so many requested changes here that I think you should abandon this pull request and start over. Make sure you're doing these things:

* `git pull` to update to the latest master (these diffs seem to be far behind)
* `git rebase` your branch
* To update a pushed branch, use `git push --force-with-lease`

Also, each commit should be one distinct change. That helps use review and cherry pick, if needed.

**Comment by briandfoy** [comment]
Okay, I think I see what happened. 

Update to the latest master of this repo then make your changes for those two files. Submit a new pull request and we should just see those two files changed. When you do work again, update master first. Always update before you start work.

    git checkout master
    git pull
    git checkout -b branch_name

After you push, delete your branches. You can use the same branch name again later, but you don't want something connected to a much earlier part of the history.

    git branch -D branch_name
    git push origin :branch_name

As such, I'm closing this PR so you can redo it.

**Comment by wpr-curly2000** [comment]
Mea Culpa Brian!

Thanks for your patience and insight!  I will get the repo updated and run the commands as you have as you have outlined.  

Thanks, Jeff.

**Pull request closed without merging**